internal/metrics/metrics.go

```diff
@@ -0,0 +1,141 @@
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	lokiv1beta1 "github.com/ViaQ/loki-operator/api/v1beta1"
+)
+
+// DefinedLimitsTypeLabel defines a label that describes the type of limits
+// imposed on the cluster
+type DefinedLimitsTypeLabel string
+
+const (
+	// Global defines a metric label for the defined usage
+	Global DefinedLimitsTypeLabel = "global"
+	// Tenant defines a metric label for the defined usage
+	Tenant DefinedLimitsTypeLabel = "tenant"
+)
+
+// LokiComponentMetricLabel defines a label that describes a component
+type LokiComponentMetricLabel string
+
+const (
+	// Querier defines a metric label for usage of a component
+	Querier LokiComponentMetricLabel = "querier"
+	// Ingester defines a metric label for usage of a component
+	Ingester LokiComponentMetricLabel = "ingester"
+	// Compactor defines a metric label for usage of a component
+	Compactor LokiComponentMetricLabel = "compactor"
+	// Distributor defines a metric label for usage of a component
+	Distributor LokiComponentMetricLabel = "distributor"
+	// QueryFrontend defines a metric label for usage of a component
+	QueryFrontend LokiComponentMetricLabel = "query_frontend"
+)
+
+var (
+	componentRestartMetric = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "lokistack_component_restart_total",
+			Help: "Number of times a component undergoes a restart",
+		},
+		[]string{"size", "component"},
+	)
```

Comment:
Why do we need this metric? IIRC there is not acceptance criteria to count restarts